src/coreComponents/physicsSolvers/multiphysics/MultiphasePoromechanicsKernel.hpp

```diff
@@ -431,100 +667,158 @@ class Multiphase :
                                     FunctionSpace::P0,
                                     DifferentialOperator::Identity,
                                     DifferentialOperator::Identity >
     (
       stack.dLocalResidualMass_dPressure,
       1.0,
-      dComponentMassContent_dPressure,
+      stack.dCompMassIncrement_dPressure,
       1.0,
       detJxW );
 
-    // --- --- Compute local mass balance residual derivatives with respect to components
+    // compute local mass balance residual derivatives with respect to components
     BilinearFormUtilities::compute< FunctionSpace::P0,
                                     FunctionSpace::P0,
                                     DifferentialOperator::Identity,
                                     DifferentialOperator::Identity >
     (
       stack.dLocalResidualMass_dComponents,
       1.0,
-      dComponentMassContent_dComponents,
+      stack.dCompMassIncrement_dComponents,
       1.0,
       detJxW );
 
-    // --- Pore volume constraint equation
-    // --- --- Local pore volume contraint residual
+
+    // Step 2: pore volume constraint equation
+
+    // compute local pore volume contraint residual
     LinearFormUtilities::compute< FunctionSpace::P0,
                                   DifferentialOperator::Identity >
     (
-      stack.localPoreVolumeConstraint,
+      stack.localResidualPoreVolConstraint,
       1.0,
-      poreVolumeConstraint,
+      stack.poreVolConstraint,
       detJxW );
 
-    // --- --- Compute local pore volume contraint residual derivatives with respect to pressure
+    // compute local pore volume contraint residual derivatives with respect to pressure
     BilinearFormUtilities::compute< FunctionSpace::P0,
                                     FunctionSpace::P0,
                                     DifferentialOperator::Identity,
                                     DifferentialOperator::Identity >
     (
-      stack.dLocalPoreVolumeConstraint_dPressure,
+      stack.dLocalResidualPoreVolConstraint_dPressure,
       1.0,
-      dPoreVolumeConstraint_dPressure,
+      stack.dPoreVolConstraint_dPressure,
       1.0,
       detJxW );
 
-    // --- --- Compute local pore volume contraint residual derivatives with respect to components
+    // compute local pore volume contraint residual derivatives with respect to components
     BilinearFormUtilities::compute< FunctionSpace::P0,
                                     FunctionSpace::P0,
                                     DifferentialOperator::Identity,
                                     DifferentialOperator::Identity >
     (
-      stack.dLocalPoreVolumeConstraint_dComponents,
+      stack.dLocalResidualPoreVolConstraint_dComponents,
       1.0,
-      dPoreVolumeConstraint_dComponents,
+      stack.dPoreVolConstraint_dComponents,
       1.0,
       detJxW );
+  }
+
+  GEOSX_HOST_DEVICE
+  GEOSX_FORCE_INLINE
+  void quadraturePointKernel( localIndex const k,
```

Comment:
This name has to be understood in the context of the base class `finiteElement::KernelBase` that has the generic `forAll` loop [here](https://github.com/GEOSX/GEOSX/blob/e9469bf176dbe673e0edd0cb6fbce37d5a4e512c/src/coreComponents/finiteElement/kernelInterface/KernelBase.hpp#L245). In this context, the name `quadraturePointKernel` makes sense since the kernel is called inside a loop over quadrature points. I think it is deliberately generic because it is called by many solvers for different physics (simple PDEs, solid mechanics, wave propagation) across the code.